spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -337,7 +337,7 @@ private void replaceManifests(
       addedManifests.forEach(rewriteManifests::addManifest);
       commit(rewriteManifests);
 
-      if (!snapshotIdInheritanceEnabled) {
+      if (formatVersion == 1 && !snapshotIdInheritanceEnabled) {
```

Comment:
Minor suggestion: can we have format version as an extra parameter for the method as following, so that we can consolidate the logic from multiple places?
```
    PropertyUtil.propertyAsBoolean(
              table.properties(),
              TableProperties.SNAPSHOT_ID_INHERITANCE_ENABLED,
              TableProperties.SNAPSHOT_ID_INHERITANCE_ENABLED_DEFAULT,
              formatVersion);
```

Reply:
I tried to generalize this in `SnapshotProducer`, I'd delay a utility call like this until we have multiple use cases. It is a bit special as if format is V2, the value must be ignored.